src/events/deposits.controller.ts

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Controller, Get, UseGuards } from '@nestjs/common';
+import { ApiOperation, ApiTags } from '@nestjs/swagger';
+import { ApiKeyGuard } from '../auth/guards/api-key.guard';
+import { DepositsService } from './deposits.service';
+import { SerializedDeposit } from './interfaces/serialized-deposit';
+
+@ApiTags('Deposit')
+@Controller('deposits')
+export class DepositsController {
+  constructor(private readonly depositService: DepositsService) {}
+
+  @ApiOperation({ summary: 'Gets the head of the chain' })
+  @UseGuards(ApiKeyGuard)
+  @Get('head')
+  async head(): Promise<{ deposit: SerializedDeposit | null }> {
+    const deposit = await this.depositService.head();
+
+    if (!deposit) {
+      return { deposit: null };
+    }
+
+    return {
+      deposit: {
+        id: deposit.id,
```

Comment:
Not a blocker, but is there any preference between putting all these fields top-level versus under a `deposit` field?

Reply:
There is a bug in the serializer in that it turns a null response into an empty object, but only if it's the root return valule.

I'm fine being consistent with the other APIs, though I usually always put data in a key because if you ever need to add another top level object to return you're screwed.

Reply:
Fixed this by using 404 instead of returning null, and removing the nested key.